### PR TITLE
readme: Optimizely DFM

### PR DESCRIPTION
**Pull request by msohailhussain**
Summary
-------

-  Added documentation for DFM
  - Initialize Optimizely using SDKKey, PollingConfigManager and datafile
  - Added property descriptions of PollingConfigManager class or it's super class.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -30,7 +30,94 @@ dashboard, please contact your Optimizely account executive.
 Using the SDK
 ~~~~~~~~~~~~~
 
-See the Optimizely `Full Stack documentation`_ to learn how to
+You can initialize the Optimizely instance in three ways: with a datafile, by providing an `sdk_key`, or by providing a Config Manager. Each method is described below.
```
Comment (aliabbasrizvi) [inline README.rst]: Link `ConfigManager` to `config_manager.py`

File: README.rst [hunk 2]
```diff
@@ -30,7 +30,94 @@ dashboard, please contact your Optimizely account executive.
...
+   a PollingConfigManager that makes an HTTP GET request to the URL (formed 
+    using your provided `sdk key` and the default datafile CDN url
+   template) to asynchronously download the project datafile at regular
+   intervals and update ProjectConfig when a new datafile is recieved. A
+   hard-coded datafile can also be provided along with the `sdk_key` that
+   will be used initially before any update.
+   ::
+
+      optimizely.Optimizely(
+        datafile=None,
```
Comment (aliabbasrizvi) [inline README.rst]: nit. I don't think our example needs to show `datafile=None`

File: README.rst [hunk 3]
```diff
@@ -30,7 +30,94 @@ dashboard, please contact your Optimizely account executive.
...
+        sdk_key='put_your_sdk_key_here'
+      )
+
+3. Initialize Optimizely by providing a Config Manager that implements a
```
Comment (aliabbasrizvi) [inline README.rst]: Instead, link to `BaseConfigManager` and ask for user to implement `BaseConfigManager`.

File: README.rst [hunk 4]
```diff
@@ -42,11 +41,11 @@ different ways as per your requirement.
       )
 
 2. Initialize Optimizely by providing an 'sdk_key'. This will initialize
-   a PollingConfigManager that makes an HTTP GET request to the URL (
-   formed using your provided sdk key and the default datafile CDN url
+   a PollingConfigManager that makes an HTTP GET request to the URL (formed 
+    using your provided `sdk key` and the default datafile CDN url
```
Comment (hillaryfraley-optimizely) [inline README.rst]: put "URL" in capital letters?

File: README.rst [hunk 5]
```diff
@@ -91,31 +88,31 @@ successful datafile poll.
...
-datafile         None                                                     Initial datafile, typically sourced from a local cached source.
+url              None                                                     URL override location used to specify custom HTTP source for the Optimizely datafile
+url_template     https://cdn.optimizely.com/datafiles/{sdk_key}.json      Parameterized datafile URL by SDK key
+datafile         None                                                     Initial datafile, typically sourced from a local cached source
 ================ ======================================================== =====================================================================================
 
 A notification signal will be triggered whenever a *new* datafile is
 fetched and Project Config is updated. To subscribe to these
-notifications you can use the
+notifications, use the
```
Comment (hillaryfraley-optimizely) [inline README.rst]: Maybe this instead?
To subscribe to these notifications, use:

**Comment by aliabbasrizvi** [comment]
@msohailhussain can you fix the PR summary?

**Comment by mikeng13** [comment]
> @msohailhussain can you fix the PR summary?

This is done. Are we good to merge in?

**Pull request merged**